### DISCUSSION TITLE
bot.telegram.sendMessage with inline keyboard and html

### DISCUSSION THREAD

**Discussion by pmioduszewski** [telegraf/telegraf#1322] [Q&A] [answered]
Hi,
I can't figure out how to send message (not as reply) with inline keyboard + HTML parsing.

I've tried something like this:
`bot.telegram.sendMessage(123456789, 'This is <b>text</b> message', {
							parse_mode: 'HTML',
							inline_keyboard: [
							[
								{
									'text': '⚡ Status',
									'callback_data': 'status',
								},
								{
									'text': '⚙ Account',
									'callback_data': 'account',
								},
							]
							]});`

Any working example?

Regards!

**Comment by wojpawlik** [comment]
TypeScript will explain the issue when you add `// @ts-check` to the very beginning of the file.

**Comment by wojpawlik** [comment, marked as answer]
```js
await ctx.telegram.sendMessage(
  123456789,
  'This is <b>text</b> message',
  { parse_mode: 'HTML', ...Markup.inlineKeyboard(...) }
)
```

**Reply by pmioduszewski** [reply]
Thx for reply,
"ctx" from where? 
I need this not as respond to some interaction with bot so I don't have "context".
Also I forgot to mention that I use JS instead of TS and working on Firebase Functions :)

I've tried this without success:
```js
	const bot = new Telegraf(functions.config().telegram_token.key);
	const keyboard = Markup.inlineKeyboard([
		[
			Markup.button.callback('⚡ Status', 'status'),
			Markup.button.callback('🙄 Help', 'help'),
		],
	]).resize();
	bot.telegram.sendMessage(uid, text_message, {parse_mode: 'HTML', inline_keyboard: keyboard});
```

AND

```js
	const bot = new Telegraf(functions.config().telegram_token.key);
	const keyboard = Markup.inlineKeyboard([
		[
			Markup.button.callback('⚡ Status', 'status'),
			Markup.button.callback('🙄 Help', 'help'),
		],
	]).resize();
	bot.telegram.sendMessage(uid, text_message, {parse_mode: 'HTML', keyboard});
```

In both cases message was sent as HTML but without keyboard.

**Reply by pmioduszewski** [reply]
Ok, so I found the solution:

```js
	const keyboard = Markup.inlineKeyboard([
		[
			Markup.button.callback('⚡ Status', 'status'),
			Markup.button.callback('🙄 Help', 'help'),
		],
	]);
	bot.telegram.sendMessage(uid, reply_message, {
		parse_mode: 'html',
		reply_markup: keyboard.reply_markup,
	});
```

😄

**Comment by vladkras** [comment]
btw, TypeScript requires uppercase 'HTML':
```
Type '"html"' is not assignable to type 'ParseMode | undefined'. Did you mean '"HTML"'?
```
